### PR TITLE
RX.h refactor, misc. clean-up / fixes

### PR DESCRIPTION
**Pull request by csurf**
big refactor of RX code, move i2c RX slave code to separate file, swap loops for memset/memcpy, misc. clean-up and fixes

### CODE REVIEW

File: openLRSng/RX.h [hunk 1]
```diff
@@ -39,50 +57,206 @@ volatile uint8_t disablePWM = 0;
...
+  }
 
-pinMask_t chToMask[PPM_CHANNELS];
-pinMask_t clearMask;
+  if ((rx_config.pinMapping[SDA_OUTPUT] == PINMAP_SDA) &&
+      (rx_config.pinMapping[SCL_OUTPUT] == PINMAP_SCL)) {
+    myI2C_init(1);
+#ifdef ENABLE_SLAVE_MODE
+    if (rx_config.flags & SLAVE_MODE) {
+      Serial.println("I am slave");
```
Comment (mikeller) [inline openLRSng/RX.h]: :-)

File: openLRSng/RX.h [hunk 2]
```diff
@@ -371,26 +877,28 @@ uint8_t bindReceive(uint32_t timeout)
...
       rfmGetPacket(rxc_buf, len);
-
-      if (rxc_buf[0] == 'b') {
-        for (uint8_t i = 0; i < sizeof(bind_data) && i < sizeof(rxc_buf); i++) {
-          *(((uint8_t*) &bind_data) + i) = rxc_buf[i + 1];
-        }
-
+      switch((char) rxc_buf[0]) {
+      case 'b': { // GET bind_data
+        // for (uint8_t i = 0; i < sizeof(bind_data); i++) {
```
Comment (mikeller) [inline openLRSng/RX.h]: Why is this commented out? It should either be removed, or have a comment as to why it is commented out.

Comment (csurf) [reply openLRSng/RX.h]: oops, forgot to remove, fixed...

**Comment by fiveangle** [comment]
minor warning...
```
The command "opt_set COMPILE_TX 0" exited with 0.
3.33s$ build_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
In file included from /home/travis/build/fiveangle/openLRSng/openLRSng/openLRSng.ino:112:0:
openLRSng/RX.h: In function 'void checkBinaryMode()':
openLRSng/RX.h:1013:40: warning: 'void checkBinaryMode()' was declared 'extern' and later 'static' [-fpermissive]
static inline void checkBinaryMode(void)
^
openLRSng/RX.h:117:6: note: previous declaration of 'void checkBinaryMode()'
void checkBinaryMode(void);
^
```

also, @mikeller - you should be able to enable TravisCI checking now without issue:

https://travis-ci.org/fiveangle/openLRSng/builds/335885669

**Comment by mikeller** [comment]
Thanks for that, @fiveangle, activated.

I saw that there are some warnings in the build:

```
In file included from /home/travis/build/openLRSng/openLRSng/openLRSng/openLRSng.ino:112:0:
openLRSng/RX.h: In function 'void checkBinaryMode()':
openLRSng/RX.h:1013:40: warning: 'void checkBinaryMode()' was declared 'extern' and later 'static' [-fpermissive]
static inline void checkBinaryMode(void)
^
openLRSng/RX.h:117:6: note: previous declaration of 'void checkBinaryMode()'
void checkBinaryMode(void);
```

Should we turn on 'warnings-as-errors'? I find that it often provides valuable 'early detection' of bad smells.

**Comment by fiveangle** [comment]
You're probably right that it would be better to catch these earlier: #182

**Pull request merged**